### ISSUE TITLE
The "Server Error (500)" is displayed if the Branch - "Name" section has no value inserted in the input field when saved

### ISSUE THREAD

**Issue by Softvision-RemusDranca** [mozilla/experimenter#1203]
**[Affected versions]:**
- Firefox Release 66.0.3 and above
- Chrome 74.0. and up

**[Affected Platforms]:** 	 	
- All Windows
- All Mac
- All Linux

**[Prerequisites]:**
- Have an experiment created on the "https://stage.experimenter.nonprod.dataops.mozgcp.net/" website.
- You are logged in your LDAP account on stage website.

**[Steps to reproduce]:**
1. Open the browser with the profile from prerequisites and navigate to "https://stage.experimenter.nonprod.dataops.mozgcp.net/".
2. Open the created experiment from prerequisites.
3. Click the "Edit" button from "Branches" section.
4. Delete the value from the "Name" input field.
5. Click the "Save Draft" button and observe the behavior.

**[Expected result]:**
- The "! This field is required" error message is displayed. 

**[Actual result]:**
- The "Server Error (500)" is displayed.

**[Notes]:**
- This issue is also reproducible if empty spaces are inserted in the "Name" input field.
- Attached is a screen recording of the issue:
![Error500](https://user-images.githubusercontent.com/40061643/56655553-ca35d380-669b-11e9-889e-c33897cd1fa2.gif)

**Comment by jaredkerim** [comment]
Ah yes this is a regression caused by https://github.com/mozilla/experimenter/issues/1129 

We should add a new test case to recreate this and then make sure that field is non-empty before we try to do the slug validation.

**Comment by Softvision-RemusDranca** [comment]
I have verified this issue and is no longer reproducible on Experimenter stage "https://stage.experimenter.nonprod.dataops.mozgcp.net/" website after the v2.28.1 stage deployment.
However, the wrong error message is displayed after saving an experiment with no value inserted in the Branch - Name section. I opened a new issue #1253 reflecting this behavior.
Tested on Windows 10x64, Mac 10.14 and Arch Linux 4.16.6 x64.
